opencensus/stats/measurement_map.py

```diff
@@ -63,10 +63,15 @@ def measure_put_attachment(self, key, value):
             If this method is called multiple times with the same key,
             only the last value will be kept.
         """
-        if key is None:
-            raise TypeError('attachment key should not be empty')
-        if value is None:
-            raise TypeError('attachment value should not be empty')
+        if self._attachments is None:
+            raise TypeError('attachments should not be empty')
```

Comment:
If `_attachments` is `None` and users want to put new attachments, it should be created as an empty dict instead of raising error.